Initial test suite

tomgeorge opened the pull request:
Initial PR to add some test coverage to this.  I've noticed that running the integration tests are rather difficult, because you need a running workspace, and CHE_MACHINE_TOKEN, which is a fairly involved process. 

Part of https://github.com/redhat-developer/rh-che/issues/1712

davidfestal commented:
Do you think you could gather all the formatting changes in a dedicated commit ?

tomgeorge commented:
Yes, I didn't realize intellij would format the whole thing

tomgeorge commented:
Ok I cleaned up some of the formatting in `AbstractAnalyticsManager` but I removed one of the `try` blocks so everything is shifted over and is difficult to read the diff.

ibuziuk commented:
@davidfestal @benoitf could you please add me to the che-incubator org, so that I could review & merge PRs ?

davidfestal commented on backend-base/src/main/resources/application.properties:
```diff
@@ -1,4 +1,6 @@
-quarkus.ssl.native=true
\ No newline at end of file
+quarkus.ssl.native=true
+che.api=http://che-che.apps-crc.testing/api
```
These 2 lines certainly make tests easier, but might bring failure at runtime. They should not be made available to the real application.

benoitf commented:
@ibuziuk added che team. I don't know why but this team was not part of this repo

ibuziuk commented on backend-base/src/test/java/org/eclipse/che/incubator/workspace/telemetry/NativeTelemetryResourceIT.java:
```diff
@@ -1,9 +1,37 @@
+                .then()
+                .statusCode(200);
+    }
 }
```
No new line

ibuziuk commented on backend-base/src/test/java/org/eclipse/che/incubator/workspace/telemetry/TelemetryResourceTest.java:
```diff
@@ -1,21 +1,49 @@
+        String response = telemetryResource.event(e);
+        assertEquals("", response);
+    }
 }
```
No new line

ibuziuk commented on backend-base/src/test/java/org/eclipse/che/incubator/workspace/telemetry/base/AbstractAnalyticsManagerTest.java:
```diff
@@ -0,0 +1,45 @@
+        assertEquals(analyticsManager.stackId, "Go");
+        assertEquals(analyticsManager.firstStart, true);
+    }
+}
```
No new line

ibuziuk commented on backend-base/src/test/resources/META-INF/beans.xml:
```diff
@@ -0,0 +1,6 @@
+       xsi:schemaLocation="
+      http://java.sun.com/xml/ns/javaee
+      http://java.sun.com/xml/ns/javaee/beans_1_0.xsd">
+</beans>
```
No new line

ibuziuk commented on backend-base/src/test/resources/mock-response.json:
```diff
@@ -0,0 +1,293 @@
+      }
+    ]
+  }
+}
```
No new line

ibuziuk commented on backend-base/src/test/java/org/eclipse/che/incubator/workspace/telemetry/base/MockBaseConfiguration.java:
```diff
@@ -0,0 +1,127 @@
+        public MockHttpRequest() {
+            mockResponse = this.getMockResponse();
+            if (mockResponse.equals("")) {
+                System.err.println("No mock response found");
```
could be addressed in a separate PR, but later we should probably add a logger

ibuziuk replied:
`org.slf4j` seems to be already available in the project btw

tomgeorge replied:
You are right, this requires a bigger refactor.  We are instantiating loggers ourselves instead of injecting them, and when we inject them we need to move the constructor logic into `@PostConstruct`.  I'll open a PR for this.

ibuziuk commented on backend-base/src/test/java/org/eclipse/che/incubator/workspace/telemetry/base/MockBaseConfiguration.java:
```diff
@@ -0,0 +1,127 @@
+            mockResponse = this.getMockResponse();
+            if (mockResponse.equals("")) {
+                System.err.println("No mock response found");
+                System.exit(1);
```
Do we really need System.exit here ?

tomgeorge replied:
I can throw a response instead

ibuziuk commented on backend-base/src/test/resources/mock-response.json:
```diff
@@ -0,0 +1,293 @@
+{
```
Just to make sure. This is a valid Che 7.x API response?

tomgeorge replied:
Yeah, this was grabbed from a running Che workspace
